rules_java_gapic/java_gapic.bzl

```diff
@@ -138,6 +140,20 @@ def java_gapic_library(
     if gapic_yaml:
         file_args_dict[gapic_yaml] = "gapic-config"
 
+    # Check the allow-list.
+    # Open this up after mixins are published, and gate on
+    # the allowlisted "mised-in" APIs present in Java.
+    if service_yaml:
+        service_yaml_in_allowlist = False
+        for keyword in SERVICE_YAML_ALLOWLIST:
+            if keyword in service_yaml:
+                service_yaml_in_allowlist = True
+                break
+        if service_yaml_in_allowlist:
+            file_args_dict[service_yaml] = "gapic-service-config"
```

Comment:
Nit: Why `gapic-service-config`? If it is because that's what Go had, I just renamed it to `api-service-config` to be more accurate.

Reply:
I brought this back from a previous PR, so it's actually from months-old code. Agree the name should be changed, I'll do that in a separate PR.